File: serverTCC/serverTCC/Models/Transferencia.cs
using System;

namespace serverTCC.Models
{
    public class Transferencia
    {
        public int Id { get; set; }

        public int ReceitaId { get; set; }
        public Movimentacao Receita { get; set; }

        public int DespesaId { get; set; }
        public Movimentacao Despesa { get; set; }
    }
}

File: client/www/js/models/transferencia.js
function Transferencia(data) {
    "use strict";
    if (!data) data = {};

    this.id = data.id;
    this.receitaId = data.receitaId;
    if (data.receita)
        this.receita = new Movimentacao(data.receita);
    this.despesaId = data.despesaId;
    if (data.despesa)
        this.despesa = new Movimentacao(data.despesa);
}

localEntities.register(Transferencia, {
    id: {
        type: 'number',
        pk: true
    },
    receitaId: {
        type: 'number',
        fk: true
    },
    receita: {
        type: 'Movimentacao',
        relationKey: 'receitaId'
    },
    despesaId: {
        type: 'number',
        fk: true
    },
    despesa: {
        type: 'Movimentacao',
        relationKey: 'despesaId'
    }
});